src/actions/mainActions.js

```diff
@@ -0,0 +1,40 @@
+import * as types from './actionTypes';
+
+export function setTitleActive() {
+    return {
+        type: types.SET_TITLE_ACTIVE
+    }
+}
+export function setGenreActive() {
+    return {
+        type: types.SET_GENRE_ACTIVE
+    }
+}
+export function setSearchValue(val) {
+    return {
+        type: types.SET_SEARCH_STR,
+        value: val
+    }
+}
+export function sortByReleaseDate() {
+    return {
+        type: types.SORT_RELEASE_DATE
+    }
+}
+export function sortByRating() {
+    return {
+        type: types.SORT_RATING
+    }
+}
+export function fetchMoviesSuccess(movies) {
+    return {
+        type: types.FETCH_MOVIES_SUCCESS,
+        movies
+    }
+}
+export function fetchMovies(url) {
+    return function(dispatch) {

```

Comment:
Another good practice is having at least 3 states for an http request, a state to say that you started a request, one for success request and another for failure.

```javascript
export const requestMovieData = () => ({ type: types.REQUEST_MOVIE_DATA });
export const receiveMovieData = (payload) => ({ type: types.RECEIVE_MOVIE_DATA, payload });
export const receiveMovieDataFailed = (err) => ({ type: types.RECEIVE_MOVIE_DATA_FAILED, payload: err, error: true });
```

Reference: https://redux.js.org/advanced/async-actions#actions

Reply:
Okay